src/Microsoft.AspNet.Http/HttpContextFactory.cs

```diff
@@ -17,13 +21,19 @@ public HttpContextFactory(IHttpContextAccessor httpContextAccessor)
         public HttpContext Create(IFeatureCollection featureCollection)
         {
             var httpContext = new DefaultHttpContext(featureCollection);
-            _httpContextAccessor.HttpContext = httpContext;
+            if (_httpContextAccessor != null)
+            {
+                _httpContextAccessor.HttpContext = httpContext;
```

Comment:
Looks like there's a proposal (added to the C# 7 and VB 15 milestone) for this: https://github.com/dotnet/roslyn/issues/1276